src/OrbitGl/LineGraphTrack.cpp

```diff
@@ -57,14 +58,36 @@ void LineGraphTrack<Dimension>::DrawSeries(PrimitiveAssembler& primitive_assembl
   std::array<float, Dimension> prev_normalized_values =
       GetNormalizedValues(curr_iterator->second, min, inverse_value_range);
 
-  bool is_first = true;
-  GraphTrackDataAggregator<Dimension> aggr(this->aggregation_mode_);
+  GraphTrackDataAggregator<Dimension> aggr;
 
   const uint32_t resolution_in_pixels =
       this->GetViewport()->WorldToScreen({this->GetWidth(), 0})[0];
   uint64_t next_pixel_start_ns = orbit_client_data::GetNextPixelBoundaryTimeNs(
       min_tick, resolution_in_pixels, min_tick, max_tick);
 
+  // Issues draws for the aggregated entry depending on `aggregation_mode_` and updates
+  // `prev_normalized_values` with last drawn values.
+  auto draw_aggregated =
+      [&](const typename GraphTrackDataAggregator<Dimension>::AccumulatedEntry& accumulated_entry,
+          bool is_last) {
+        // First dtaw the entry for the max values.
+        DrawSingleSeriesEntry(primitive_assembler, accumulated_entry.start_tick,
```

Comment:
You are assuming that the max_value is before the min_value. To be more precise I would prefer to store the min_value_timestamp and the max_value_timestamp and use the order provided by the timestamps.

I think it shouldn't matter too much, so I leave addressing this issue to you, but I can generate a border case with an undesired behaviour. Let me know if you want to talk offline.

Reply:
Great catch, thank you so much for the suggestion and the followup discussion!